### PR TITLE
added log to on_ip_change and notify if ip_notifier is not supported

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/ip_notifier.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ namespace libtorrent
 				{ cb(error_code(err, system_category())); });
 		}
 #else
-		TORRENT_UNUSED(cb);
+		cb(make_error_code(boost::system::errc::not_supported));
```
Comment (aldenml) [inline src/ip_notifier.cpp]: @ssiloti do you think this is a good idea?

File: src/session_impl.cpp [hunk 2]
```diff
@@ -1763,6 +1763,10 @@ namespace aux {
 
 	void session_impl::on_ip_change(error_code const& ec)
 	{
+#ifndef TORRENT_DISABLE_LOGGING
+		session_log("received ip change from internal ip_notifier");
```
Comment (arvidn) [inline src/session_impl.cpp]: I would prefer a single log, either an error or a success.

**Comment by arvidn** [comment]
looks reasonable to me

**Comment by aldenml** [comment]
@arvidn Are you open to add
```
target_link_libraries(torrent-rasterbar "-framework CoreFoundation" "-framework SystemConfiguration")
```
in a possible support for ip change notification in macOS?

**Comment by arvidn** [comment]
yes. But I think it's important to be added to the makefile and Jamfile too.

File: src/ip_notifier.cpp (hunk 1 above)
Comment (ssiloti) [inline src/ip_notifier.cpp]: LGTM. It should probably do the same thing in the simulator case for consistency.

Comment (aldenml) [inline src/ip_notifier.cpp]: I hope you don't mind another PR for that, CI is green already :)

btw, I think I have a solution for macOS

**Pull request merged**